imap/mboxlist.c

```diff
@@ -1049,7 +1047,10 @@ static int mboxlist_update_raclmodseq_wrapper(const char *acluser,
     int r = 0;
     int i;
     for (i = 0; i < strarray_size(members); i++) {
-        r = mboxlist_update_raclmodseq(strarray_nth(members, i), touched_users);
+	const char *member = strarray_nth(members, i);
+        if (strarray_find(touched_users, member, 0) >= 0) return 0;
```

Comment:
You're quite right.  Also can use strarray_append rather than strarray_add since we've literally just done the test that starray_add will run in a second.  Fixed.